CleanWeather/Scenes/CityHistorical/CityHistoricalWorker.swift

```diff
@@ -23,9 +25,28 @@ final class CityHistoricalWorkerImpl {
 
 extension CityHistoricalWorkerImpl: CityHistoricalWorker {
 
-    func fetchCityHistoricalWeather(id: String, date: Date, completion: FetchHistoricalCompletion?) {
+    func fetchCityHistoricalWeather(cityDetails: CityWeather, date: Date, completion: FetchHistoricalCompletion?) {
         let convertedDate = unixFormatDate(date: date)
-        networking.fetchHistoricalWeatherForCity(id: id, date: convertedDate, completion: completion)
+        let coordinates = "\(cityDetails.latitude),\(cityDetails.longitude)"
+        let id = cityDetails.id
+        let name = cityDetails.city
+        networking.fetchHistoricalWeatherForCity(coordinates: coordinates, date: convertedDate) { result in
+            switch result {
+            case .success(let apiResponse):
+                print(apiResponse)
+                guard let temp = apiResponse.currently.temperature else {
```

Comment:
check for icon too

Reply:
`icon: apiResponse.currently.icon ?? ""`
I didn't check for icon and provided empty String because DarkSkyAPI missing a lot of icons for past weather conditions while still has temperature values. For that occasion user still has temperature data for historical date and WeatherSymbol enum returns "NoSign" icon.
`return WeatherSymbol(rawValue: icon) ?? .isEmpty`

Reply:
If I guard icon optional with .failure user will not get lot of temp data which api did send back for given date. For example Katowice has icons only since year 2008 while temperature data for as far as 1999.